h/models.py

```diff
@@ -115,13 +115,25 @@ def __acl__(self):
     }
     __analysis__ = {
         'filter': {
-            'uri': {
+            'uri_index': {
                 'type': 'pattern_capture',
                 'preserve_original': '1',
                 'patterns': [
                     '([^\\/\\?\\#\\.]+)',
                     '([a-zA-Z0-9]+)(?:\\.([a-zA-Z0-9]+))*',
                     '([a-zA-Z0-9-]+)(?:\\.([a-zA-Z0-9-]+))*',
+                    '://(.+)',
+                    '://(.+)\\?',
+                    '://(.+)\\#'
+                ]
+            },
+            'uri_search': {
+                'type': 'pattern_capture',
+                'preserve_original': '1',
+                'patterns': [
+                    '://(.+)',
+                    '://(.+)\\?',
```

Comment:
> Yes, this can produce loading orphan annotation

Orphan annotations are the least of our problems. Real false matches are worse. We can easily find the same piece of text in different documents, talking about different things. And then we would anchor the annotations to place where they don't belong to.

> but at the same time we should treat ```http://site.com/content?mobile=0``` and ```http://site.com/content?mobile=1``` the same in the sake of annotations.

That's why we have the support for `canonical-uri` and friends - right? Those should be linked using some meta tags.